File: src/main/kotlin/com/meow/meowchatting/user/command/domain/Friend.java
package com.meow.meowchatting.user.command.domain;

import com.meow.meowchatting.common.base.AbstractBaseUserByEntity;

import jakarta.persistence.AttributeOverride;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "friend")
@Getter @NoArgsConstructor
@AttributeOverride(name = "id", column = @Column(name = "friend_id"))
public class Friend extends AbstractBaseUserByEntity {

	@Column(name = "user_id", nullable = false)
	private Long userId;

	@Column(name = "friend_user_id", nullable = false)
	private Long friendUserId;

	@Column(name = "friend_name", nullable = false, length = 30)
	private String friendName;

	@Column(name = "is_blocked", nullable = false, columnDefinition = "TINYINT(1) DEFAULT 0")
	private Boolean isBlocked = false;

	/**
	 *  친구 이름 변경
	 */
	public void updateFriendName(String friendName) {
		this.friendName = friendName;
	}

}

File: src/main/kotlin/com/meow/meowchatting/service/user/domain/Friend.kt
package com.meow.meowchatting.service.user.domain

import com.meow.meowchatting.service.base.AbstractBaseEntity
import jakarta.persistence.*
import lombok.Getter
import lombok.Setter
import java.time.LocalDateTime

@Setter
@Getter
@Entity
@Table(name = "friend")
@AttributeOverride(name = "id", column = Column(name = "friend_id"))
class Friend : AbstractBaseEntity() {

    @Column(name = "user_id", nullable = false)
    var userId: Long = 0

    @Column(name = "friend_user_id", nullable = false)
    var friendUserId: Long = 0

    @Column(name = "friend_name", nullable = false)
    var friendName: String = ""

    @Column(name = "is_blocked", nullable = false)
    var isBlocked: Boolean = false

    @Column(name = "deleted_at", nullable = false)
    var deletedAt: LocalDateTime = LocalDateTime.now()

}
